framework/include/userobject/NearestPointBase.h

```diff
@@ -79,51 +79,36 @@ class NearestPointBase : public ElementIntegralVariableUserObject
    * @param p The point.
    * @return The UserObject closest to p.
    */
-  UserObjectType * nearestUserObject(const Point & p) const;
+  std::shared_ptr<UserObjectType> nearestUserObject(const Point & p) const;
```

Comment:
Well I guess it's time to start talking about smart pointers. I just finished reading the entire new Scott Meyer's book on C++11 and now I'm not so sure we should start sprinkling shared pointers throughout our code. He has a very good read about shared pointers vs unique pointers. I can't do the whole discussion justice here but we need to chat about it as a team so I'm going to open a ticket to start the discussion (watch for it shortly). Let's hold off on merging this until after we've had our discussion.